### PR TITLE
feat: allow throwing external redirects

### PR DESCRIPTION
**Pull request by schiller-manuel**

### CODE REVIEW

File: e2e/react-router/basic-file-based/tests/redirect.spec.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { test } from '@playwright/test'
+
+test.describe('redirects', () => {
+  test.describe('navigating to a route that redirects', () => {
+    test('internal redirect', async ({ page }) => {
+      await page.goto('/')
+      await page.getByRole('link', { name: 'redirect-internal' }).click()
+      await page.waitForURL('http://localhost:3001/posts')
```
Comment (SeanCassiere) [inline e2e/react-router/basic-file-based/tests/redirect.spec.ts:8]: The tests need to have `expect`s.
```
expect(page.url()).toBe('...')
```

**Review by SeanCassiere — COMMENTED** [review]
image

@schiller-manuel the redirects in the router side of things are working as expected.
This however fails for Start, when a `redirect` is `thrown`/`returned` from `createServerFn`.
You'll likely want to take out lines 178-189 in `start/server-handler/index.tsx` and resolve this in the router itself.

Edit: Also, you'll want to add a test to make sure redirects from `createServerFn` calls are being handled.

File: e2e/react-router/basic-file-based/tests/redirect.spec.ts (hunk 1 above)
Comment (schiller-manuel) [reply e2e/react-router/basic-file-based/tests/redirect.spec.ts]: done

**Comment by schiller-manuel** [comment]
i had to reopen it here https://github.com/TanStack/router/pull/2429

**Pull request merged**